channeldb/kvdb/etcd/readwrite_bucket.go

```diff
@@ -46,44 +46,23 @@ func (b *readWriteBucket) NestedReadBucket(key []byte) walletdb.ReadBucket {
 // is nil, but it does not include the key/value pairs within those
 // nested buckets.
 func (b *readWriteBucket) ForEach(cb func(k, v []byte) error) error {
-	prefix := makeValuePrefix(b.id)
-	prefixLen := len(prefix)
+	prefix := string(b.id)
 
 	// Get the first matching key that is in the bucket.
-	kv, err := b.tx.stm.First(string(prefix))
+	kv, err := b.tx.stm.First(prefix)
```

Comment:
👍  easier to follow now with sub-buckets and keys being iterated over in a single pass

Reply:
Yes, it's a bit cleaner now.